watertap/core/util/scaling.py

```diff
@@ -0,0 +1,149 @@
+        lb_scale : the lower bound scale which adjusts the magnitude for the
+            fixed inlet flows of the process (default = 1e-2)
+        ub_scale : the upper bound scale which adjusts the magnitude for the
+            fixed inlet flows of the process (default = 1e2)
+        tol : tolerance for the change in magnitude of scaled variables between
+            the lower and upper bounds (default = 1e3), should always be smaller
+            than the difference in magnitude of lb_scale and up_scale
+        zero: magnitude that is considered to be zero, scaled variables with a value of
+            zero are okay, and not reported (default = 1e-10)
+
+    Yields:
+        string denoting what step failed, variable name as string,
+            sensitivity of sv between lower and upper solutions
+    """
+
+    # TODO: Check DOF is 0
+
+    sv_hist = {}
+
+    for scale in [lb_scale, ub_scale]:
+
+        temp_blk = (
+            blk.clone()
+        )  # clone flowsheet to re-solve at conditions supplied by scale
+
+        # loop through all vars to wipe previous sf and reset dsf considering new scale
+        for (var_name, var_index), var_obj in temp_blk.component_data_iterindex(
+            ctype=pyo.Var, active=True, descend_into=True
+        ):
+
+            unset_scaling_factor(
+                var_obj
+            )  # remove prior sf which are reestablished on init
+
+            # if the flow term has a default scaling factor and is fixed, as standard by flowsheets,
+            # overwrite the dsf to new scale magnitude
+            # TODO: Check multiple inlet ports are handled correctly
+            if (
+                blk.fs.properties.get_default_scaling(var_name, index=var_index)
+                is not None
+                and "flow" in var_name
+                and var_obj.fixed
+            ):
+
+                # overwrite old dsf with one considering the new scale
+                dsf = blk.fs.properties.get_default_scaling(var_name, index=var_index)
+                temp_blk.fs.properties.set_default_scaling(
+                    var_name, dsf * scale**-1, index=var_index
+                )
+
+                # overwrite the fixed variable value
+                var_obj.fix(var_obj * scale)
+
+                # TODO: Determine how to consider user specified sf, e.g. ro.area
+
+        # resolve the model at the new scale
+        calculate_scaling_factors(temp_blk)
+        temp_blk.fs.unit.initialize(outlvl=idaes.logger.ERROR)
+        results = solver.solve(temp_blk)
+
+        # ensure model solves wrt new scale
+        if not pyo.check_optimal_termination(results):
+            yield "Failed run on", scale, "scale"
+            break
+
+        # check variable scaling wrt new scale
+        for b in list(badly_scaled_var_generator(temp_blk, zero=zero)):
+            yield f"badly scaled variable for scaled flow of {scale} ", b[0].name, b[1]
```

Comment:
```suggestion
        for var, scaled_value in badly_scaled_var_generator(temp_blk, zero=zero):
            yield f"badly scaled variable for scaled flow of {scale} ", var.name, scaled_value
```
No need to create an intermediate list, and tuple unpacking makes the intent clear